### DISCUSSION TITLE
Adding a recipe by taking a photo and using AI to convert it

### DISCUSSION THREAD

**Discussion by pgale** [mealie-recipes/mealie#3899] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

(There is another very old feature request to do this but using non-AI OCR. With the advances in AI technology, I thought this warranted a new feature request as the majority of the processing can now be done by AI so *might* be easier to implement)

### Please provide a concise description of the problem that would be addressed by this feature.

I have so many old hand written or typed recipes that I would like to get into Mealie. I also have photos of recipes and a stack of recipe books that I would like to get select recipes into Mealie. 

### Please provide a concise description of the feature that would resolve your issue.

I see this working as a simple photo upload (or Live Photo import if an iOS app does eventually get written) and the backend uses AI to process and import the recipe with an image if available. 

I have already successfully used chatGPT to achieve part of this. The OCR works extremely well on both images of printed text and handwritten text. I asked chatGPT to create a recipe with ingredients and method and it did it flawlessly. I then had to manually cut and paste the resultant text into Mealie. It would seem that this is the part that would need the most work in getting repeatable results. 

I would like to see an automated process that handles the interaction with openAI and imports the recipe in the right format.

The process would pull out any recipe image from the photo and allow the user to select one to use as the recipe image.

This would be a hugely beneficial feature for many users. Anyone that has cookbooks or an archive of recipes would love this!

### Please consider and list out some caveats or tradeoffs made in your design decision

Only downsides I can see is having to have an openAI paid account (as with current AI functionality) and the need to take a photo of each recipe, sometimes over multiple pages. 

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
The newest nightly has the ability to import via image using OpenAI: https://github.com/mealie-recipes/mealie/pull/3974; it can also optionally translate the recipe for you (with varying success depending on which model you use)

**Comment by Binocularbath** [comment]
very cool idea, id be interested in this too

**Reply by Kuchenpirat** [reply]
This is already in mealie.
All you have to do is provide an openai api key
